networkx/generators/random_graphs.py

```diff
@@ -730,34 +745,39 @@ def dual_barabasi_albert_graph(n, m1, m2, p, seed=None):
     elif p == 0:
         return barabasi_albert_graph(n, m2, seed)
 
-    # Add max(m1,m2) initial nodes (m0 in barabasi-speak)
-    G = empty_graph(max(m1, m2))
+    if initial is None:
+        # Default initial graph : empty graph on max(m1, m2) nodes
+        G = star_graph(max(m1, m2))
+    else:
+        if len(initial) < max(m1, m2) or len(initial) > n:
+            raise nx.NetworkXError(
+                f"Barabási–Albert initial graph must have between "
+                f"max(m1, m2) = {max(m1, m2)} and n = {n} nodes"
+            )
+        G = initial.copy()
+
     # Target nodes for new edges
-    targets = list(range(max(m1, m2)))
+    targets = list(G.nodes())
```

Comment:
```suggestion
    targets = list(G)
```
Or if you prefer `list(G.nodes)`